casper-server/src/lua/tasks.rs

```diff
@@ -92,10 +92,14 @@ fn spawn_task(lua: &Lua, arg: Value) -> Result<StdResult<TaskHandle, String>> {
     let (join_handle_tx, join_handle_rx) = oneshot::channel();
 
     let mut name = None;
+    let mut timeout = None;
     let handler = match arg {
         Value::Function(task_fn) => task_fn.into_owned(),
         Value::Table(params) => {
             name = params.get::<_, Option<String>>("name")?;
+            timeout = params
+                .get::<_, Option<u64>>("timeout")?
+                .map(tokio::time::Duration::from_millis);
```

Comment:
Probably would be better to use seconds instead of millis (everywhere else timeouts are in seconds)